kafka/producer.go

```diff
@@ -368,6 +382,8 @@ func (p *Producer) Close() {
 	p.handle.cleanup()
 
 	C.rd_kafka_destroy(p.handle.rk)
+
+	p.isClosed = true
```

Comment:
this Close() should also check that isClosed is not already done, and fix that race condition